src/azure-cli-core/azure/cli/core/extension/__init__.py

```diff
@@ -22,7 +22,7 @@
 EXTENSIONS_DIR = os.path.expanduser(_CUSTOM_EXT_DIR) if _CUSTOM_EXT_DIR else os.path.join(GLOBAL_CONFIG_DIR,
                                                                                           'cliextensions')
 DEV_EXTENSION_SOURCES = _DEV_EXTENSION_SOURCES.split(',') if _DEV_EXTENSION_SOURCES else []
-EXTENSIONS_SYS_DIR = os.path.expanduser(_CUSTOM_EXT_SYS_DIR) if _CUSTOM_EXT_SYS_DIR else os.path.join(get_python_lib(), 'azure-cli-extensions')
+EXTENSIONS_SYS_DIR = os.path.expanduser(_CUSTOM_EXT_SYS_DIR) if _CUSTOM_EXT_SYS_DIR else os.path.join(get_path("purelib"), 'azure-cli-extensions')
```

Comment:
Also see https://github.com/python/cpython/issues/85454#issuecomment-1093877038

Reply:
What's the difference between 'purelib' and 'platlib'? Will some extensions exist in platlib only, such as preinstalled extensions in cloudshell? Could you help verify it?

Reply:
See https://github.com/Azure/azure-cli/pull/28796#discussion_r1575584547